tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,296 @@
+    Types.ContractSet proxies;
+
+    uint256 l2OutputOracleStartingTimestamp;
+
+    uint256 xdmSenderSlotNumber;
+
+    /// @notice Sets up the contract.
+    function setUp() public override {
+        super.setUp();
+
+        proxies = _getContractSet();
+
+        l2OutputOracleStartingTimestamp = cfg.l2OutputOracleStartingTimestamp();
+
+        // Read the slot number for the xDomainMsgSender in the L1CrossDomainMessenger
+        try vm.readFile(
+            string.concat(
+                vm.projectRoot(),
+                "/lib/optimism/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json"
+            )
+        ) returns (string memory data) {
+            xdmSenderSlotNumber = stdJson.readUint(data, string(abi.encodePacked("$.[13].slot")));
+        } catch {
+            revert(
+                "Failed to read xDomainMsgSender slot number in lib/optimism/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json"
+            );
+        }
+    }
+
+    /// @notice The name of the script
+    function name() public pure override returns (string memory) {
+        return "SignFromJsonWithAssertions";
+    }
+
+    /// @notice Asserts that the SystemConfig is setup correctly
+    function checkSystemConfig() internal view {
+        console.log("Running chain assertions on the SystemConfig");
+
+        require(proxies.SystemConfig.code.length != 0, "200");
+
+        SystemConfig config = SystemConfig(proxies.SystemConfig);
+
+        ResourceMetering.ResourceConfig memory resourceConfig = config.resourceConfig();
+
+        require(config.owner() == cfg.finalSystemOwner(), "300");
+        require(config.overhead() == cfg.gasPriceOracleOverhead(), "400");
+        require(config.scalar() == cfg.gasPriceOracleScalar(), "500");
+        require(config.batcherHash() == bytes32(uint256(uint160(cfg.batchSenderAddress()))), "600");
+        require(config.gasLimit() == uint64(cfg.l2GenesisBlockGasLimit()), "700");
+        require(config.unsafeBlockSigner() == cfg.p2pSequencerAddress(), "800");
+        // Check _config
+        ResourceMetering.ResourceConfig memory rconfig = Constants.DEFAULT_RESOURCE_CONFIG();
+
+        require(resourceConfig.maxResourceLimit == rconfig.maxResourceLimit, "900");
+        require(resourceConfig.elasticityMultiplier == rconfig.elasticityMultiplier, "1000");
+        require(resourceConfig.baseFeeMaxChangeDenominator == rconfig.baseFeeMaxChangeDenominator, "1100");
+        require(resourceConfig.systemTxMaxGas == rconfig.systemTxMaxGas, "1200");
+        require(resourceConfig.minimumBaseFee == rconfig.minimumBaseFee, "1300");
+        require(resourceConfig.maximumBaseFee == rconfig.maximumBaseFee, "1400");
+        // Depends on start block being set to 0 in `initialize`
+        uint256 cfgStartBlock = cfg.systemConfigStartBlock();
+        require(config.startBlock() == (cfgStartBlock == 0 ? block.number : cfgStartBlock), "1500");
```

Comment:
We should expect it to be set to an exact number here, we can remove the ternary